python-stdlib/fnmatch/fnmatch.py

```diff
@@ -9,11 +9,18 @@
 The function translate(PATTERN) returns a regular expression
 corresponding to PATTERN.  (It does not compile it.)
 """
-import os
-import os.path
 import re
 
-# import functools
+
+def normcase(s):
```

Comment:
Since this actually doesn't do anything, and is local to this file, wouldn't it make more sense to then just remove all calls to it here?

Also this might break usage / hurt CPython compatibility: I have an `os.path` which does implement normcase the CPython way but this now won't be used anymore.

Reply:
Fair point, I was keen to remove the hard-coded need for a dependency because installing them is pretty dicey at the moment, and `os.path` can't be included directly from the micropython-lib folder with the `MICROPYPATH` env because it's a split package with `os`.
For situations where it's installed manually however, it makes sense to use it - so I've updated this change with a try/except ImportError fallback.

Reply:
Ok that's a nice workaround